### PR TITLE
Redirection

### LINKED ISSUE #2: Implement popup vs. redirect loginStyles
Developer should be able to specify which type to use. Notably, popups don't work in WebViews and Safari on iOS 8.

### PR DESCRIPTION
**Pull request by jswarren**
Implemented redirection workflow using migrations to save state during Shibboleth login. Resolves #2.

### CODE REVIEW

File: server/saml_server.js [hunk 1]
```diff
@@ -71,9 +71,8 @@ var cleanResponseForMongo = function (samlResponse) {
   return profile;
 }
 
-var updateUserProfile = function (profile) {
-  Meteor.users.update({email: samlResponse.profile.email}, 
-    {$set: {email: samlResponse.profile.email, profile: profile}}, {upsert: true});
+var updateUserProfile = function (user, profile) {
+  Meteor.users.update(user, { $set: {profile: profile} }, {upsert:true});
```
Comment (athyuttamre) [inline server/saml_server.js]: I don't think we're creating new users here. Line 44 finds a user with the email, and passes the user object to this function. If the user object doesn't exist though, the value of `user` in this function is `{}`.

File: server/saml_server.js [hunk 2]
```diff
@@ -39,10 +39,10 @@ init();
 
 Accounts.registerLoginHandler(function (loginRequest) {
   if (loginRequest.credentialToken && loginRequest.saml) {
-    var profile = Accounts.saml.retrieveCredential(loginRequest.credentialToken);
+    var profile = Accounts.saml.retrieveProfile(loginRequest.credentialToken);
```
Comment (athyuttamre) [inline server/saml_server.js]: Are Mongo calls on the server synchronous? If not, should we be putting the rest of the code in a callback to these?

File: server/saml_server.js (hunk 1 above)
Comment (jswarren) [inline server/saml_server.js]: Good catch. Not sure what I thought that was doing.

File: server/saml_server.js (hunk 2 above)
Comment (jswarren) [inline server/saml_server.js]: Yes, the server has access to a [synchronous API](http://guide.meteor.com/collections.html#server-collections).

Comment (athyuttamre) [inline server/saml_server.js]: Perfect! Also wow, I've never come across that guide before. Seems super useful.

**Pull request merged**